src/query/plan/operator.cpp

```diff
@@ -2808,9 +2809,15 @@ void SetPropertiesOnRecord(TRecordAccessor *record, const TypedValue &rhs, SetPr
     }
     case TypedValue::Type::Map: {
       PropertiesMap new_properties;
-      for (const auto &[prop_id, prop_value] : rhs.ValueMap()) {
-        auto key = context->db_accessor->NameToProperty(prop_id);
-        new_properties.emplace(key, prop_value);
+      for (const auto &[string_key, value] : rhs.ValueMap()) {
+        storage::PropertyId property_id;
+        if (auto it = cached_name_id.find(std::string(string_key)); it == cached_name_id.end()) {
+          property_id = context->db_accessor->NameToProperty(string_key);
+          cached_name_id.emplace(string_key, property_id);
+        } else {
+          property_id = it->second;
+        }
```

Comment:
see if we need to reverse this if, depending on the frequency of hitting the cache vs. having to go to the storage